### ISSUE TITLE
ERROR  new is not provided with the correct number of arguments

### ISSUE THREAD

**Issue by lendle** [tonyhffong/Lint.jl#38]
I'm calling `new()` without args in an inner constructor to avoid allocating/defining some fields until later. This causes Lint to announce an error. As far as I know this is the only way to do not define all fields when constructing a type.

I'm not really sure what Lint should do here, but I think it is not necessarily wrong to use `new()` without args. Thoughts?

``` julia
lintstr("type A  x::Vector ; A() = new() end")
1-element Array{LintMessage,1}:
                 none [A                   ]    3 ERROR  new is not provided with the correct number of arguments
```
